README.md

```diff
@@ -0,0 +1,61 @@
+# An HTTP (http/1.1 or h2) client for MirageOS
+
+This little library provides an HTTP client which can be usable inside a
+unikernel/[MirageOS][mirage]. It follows the same API as
+[http-lwt-client][http-lwt-client] which is pretty simple and uses:
+- [happy-eyeballs][happy-eyeballs] to resolve domain-name
+- [ocaml-tls][ocaml-tls] for the TLS layer
+- [paf][paf] for the HTTP protocol
+
+This library wants to be easy to use and it is associated to a MirageOS
+_device_ in order to facilite `functoria` to compose everything (mainly the
+TCP/IP stack) according to the user's target and give a _witness_ so as to
+be able to allocate a new connection to a peer and process the HTTP flow.
+
+## How to use it?
+
+First, you need to describe a new `http_client` device:
+```ocaml
+open Mirage
+
+type http_client = HTTP_client
+let http_client = typ HTTP_client
+
+let http_client =
+  let connect _ modname = function
+    | [ _pclock; _tcpv4v6; ctx ] ->
+      Fmt.str {ocaml|%s.connect %s|ocaml} modname ctx
+    | _ -> assert false in
+  impl ~connect "Http_mirage_client.Make"
+    (pclock @-> tcpv4v6 @-> git_client @-> http_client)
+```
+
+Then, you can decide how to construct such device:
+```ocaml
+let stack = generic_stackv4v6 default_network
+let dns   = generic_dns_client stack
+let tcp   = tcpv4v6_of_stackv4v6 stack
+
+let http_client =
+  let happy_eyeballs = git_happy_eyeballs stack dns
```

Comment:
Maybe add a comment why we use `git_happy_eyeballs` as I think that could look like a mistake to many.

Reply:
Yes, I will put a comment to explain but I will surely update the `mirage` tool to provide something like `mimic_happy_eyeballs` instead of `git_happy_eyeballs`.